gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+        ).order_by(hl.desc("n")).show(n_rows, n_cols)
+
+    logger.info(
+        "Checking distributions of filtered variants amongst variant filters..."
+    )
+
+    new_filters_dict = {
+        "frac_vqsr": hl.agg.fraction(t.filters.contains("AS_VQSR")),
+        "frac_vqsr_only": hl.agg.fraction(
+            t.filters.contains("AS_VQSR") & (t.filters.length() == 1)
+        ),
+    }
+    _filter_agg_order(
+        t, {"is_filtered": t.is_filtered}, extra_filter_checks=new_filters_dict
+    )
+
+    logger.info("Checking distributions of variant type amongst variant filters...")
+    _filter_agg_order(
+        t, {"allele_type": t.info.allele_type}, extra_filter_checks=new_filters_dict
+    )
+
+    logger.info(
+        "Checking distributions of variant type and region type amongst variant filters..."
+    )
+    _filter_agg_order(
+        t,
+        {
+            "allele_type": t.info.allele_type,
+            "in_problematic_region": t.in_problematic_region,
+        },
+        50,
+        140,
+        extra_filter_checks=new_filters_dict,
+    )
+
+    logger.info(
+        "Checking distributions of variant type, region type, and number of alt alleles amongst variant filters..."
+    )
+    _filter_agg_order(
+        t,
+        {
+            "allele_type": t.info.allele_type,
+            "in_problematic_region": t.in_problematic_region,
+            "n_alt_alleles": t.info.n_alt_alleles,
+        },
+        50,
+        140,
+        extra_filter_checks=new_filters_dict,
+    )
+
+
+def histograms_sanity_check(
+    t: Union[hl.MatrixTable, hl.Table], verbose: bool, hists: List[str] = HISTS
+) -> None:
+    """
+    Check the number of variants that have nonzero values in their n_smaller and n_larger bins of quality histograms (both raw and adj).
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param bool verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param List[str] hists: List of variant annotation histograms.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    for hist in hists:
+        for suffix in ["", "raw"]:
+            if suffix == "raw":
+                logger.info("Checking raw qual hists...")
+                hist = f"{hist}_{suffix}"
+            else:
+                logger.info("Checking adj qual hists...")
+
+            # Check subfield == 0
+            generic_field_check(
+                t,
+                cond_expr=(t.info[f"{hist}_n_smaller"] != 0),
+                check_description=f"{hist}_n_smaller == 0",
+                display_fields=[f"info.{hist}_n_smaller"],
+                verbose=verbose,
+            )
+            if hist not in [
+                "dp_hist_alt",
+                "dp_hist_all",
+            ]:  # NOTE: DP hists can have nonzero values in n_larger bin
```

Comment:
Minor since the list is only two elements long, but checking existence in a set is generally faster, unless there's a need for order.
```suggestion
            if hist not in set([
                "dp_hist_alt",
                "dp_hist_all",
            ]):  # NOTE: DP hists can have nonzero values in n_larger bin
```

Reply:
Yeah, I totally blanked on the normal format for a set. That's probably the recommended way.